packages/react-query/src/utils/inferReactQueryProcedure.ts

```diff
@@ -0,0 +1,53 @@
+import { TRPCClientErrorLike } from '@trpc/client';
+import {
+  AnyMutationProcedure,
+  AnyProcedure,
+  AnyQueryProcedure,
+  AnyRouter,
+  inferProcedureInput,
+  inferProcedureOutput,
+} from '@trpc/server';
+import { UseTRPCMutationOptions, UseTRPCQueryOptions } from '../shared';
+
+type InferQueryOptions<
+  TProcedure extends AnyProcedure,
+  TPath extends string,
+> = Omit<
+  UseTRPCQueryOptions<
+    TPath,
+    inferProcedureInput<TProcedure>,
+    inferProcedureOutput<TProcedure>,
+    inferProcedureOutput<TProcedure>,
+    TRPCClientErrorLike<TProcedure>
+  >,
+  'select'
+>;
+
+type InferMutationOptions<TProcedure extends AnyProcedure> =
+  UseTRPCMutationOptions<
+    inferProcedureInput<TProcedure>,
+    TRPCClientErrorLike<TProcedure>,
+    inferProcedureOutput<TProcedure>
+  >;
+
+type inferReactQueryProcedureOptions<
+  TRouter extends AnyRouter,
+  TPath extends string = '',
+> = {
+  [TKey in keyof TRouter['_def']['record']]: TRouter['_def']['record'][TKey] extends infer TRouterOrProcedure
+    ? TRouterOrProcedure extends AnyRouter
+      ? inferReactQueryProcedureOptions<
+          TRouterOrProcedure,
+          `${TPath}${TKey & string}.`
+        >
+      : TRouterOrProcedure extends AnyProcedure
```

Comment:
this extra check here shouldn't be necessary since we're checking above if its a router or procedure. not sure what effects on perf it might have but can simplify it down one level